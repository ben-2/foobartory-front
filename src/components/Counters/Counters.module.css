.fooBarWrapper {
  width: 100%;
}
.fooAndBarWrapper {
  margin-top: 5px;
  display: inline-block;
  width: 100%;
}
.fooCounter {
  display: inline-block;
  width: 190px;
  margin-right: 30px;
}

.fooCounterHeader {
  height: 25px;
}

.fooCounterContent {
  height: 17px;
}

.barCounter {
  display: inline-block;
  width: 190px;
  margin-left: 30px;
}

.barCounterHeader {
  height: 25px;
}

.barCounterContent {
  height: 17px;
}

.fooBarCounter {
  display: inline-block;
  width: 220px;
}

.fooBarCounterHeader {
  height: 30px;
}

.fooBarCounterContent {
  height: 22px;
}