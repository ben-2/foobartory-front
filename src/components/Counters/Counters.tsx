import styles from './Counters.module.css';

export function Counters() {
  return (
    <>
      Counters
    </>
  );
}
