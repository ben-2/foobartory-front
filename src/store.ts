import { createStore, Action, action, Thunk, thunk } from 'easy-peasy';
import { IRobotAttributes } from './store.type';
export interface IStoreModel {
  lang: string;
  setLang: Action<IStoreModel, string>;
  isGameStarted: boolean;
  isGameEnded: boolean;
  startGame: Action<IStoreModel>;
  endGame: Action<IStoreModel>;
  robotsConfiguration: IRobotAttributes[];
  countFoo: number;
  countBar: number;
  countFooBar: number;
  setIsRobotAvailable: Thunk<IStoreModel, { id: number, value: boolean }>;
  updateRobotConfiguration: Action<IStoreModel, IRobotAttributes[]>;
  setRobotActivity: Thunk<IStoreModel, { id: number, value: 'foo' | 'bar' | 'foobar' | 'robot' | '' }>;
  incrementFooCounter: Action<IStoreModel, number>;
  incrementBarCounter: Action<IStoreModel, number>;
  incrementFooBarCounter: Action<IStoreModel, number>;
  decrementFooCounter: Action<IStoreModel, number>;
  decrementBuyRobot: Action<IStoreModel, { foo: number, foobar: number }>;
  decrementBarCounter: Action<IStoreModel, number>;
  addRobot: Thunk<IStoreModel>;
}

const model: IStoreModel = {
  lang: 'fr',
  setLang: action((state, payload) => {
    state.lang = payload;
  }),
  isGameStarted: false,
  isGameEnded: false,
  startGame: action((state) => {
    state.isGameStarted = true;
  }),
  endGame: action((state) => {
    state.isGameEnded = true;
  }),
  robotsConfiguration: [
    {
      id: 1,
      currentActivity: '',
      isRobotAvailable: true,
      previousActivity: '',
    },
    {
      id: 2,
      currentActivity: '',
      isRobotAvailable: true,
      previousActivity: '',
    }
  ],
  countFoo: 0,
  countBar: 0,
  countFooBar: 0,
  setIsRobotAvailable: thunk((actions, payload, { getState }) => {
    const storeState = getState();
    const currentRobotConf = storeState.robotsConfiguration.map((robot) => {
      const currentRobot = robot;
      if (robot.id === payload.id) {
        currentRobot.isRobotAvailable = payload.value;
      }
      return currentRobot;
    });
    actions.updateRobotConfiguration(currentRobotConf);
  }),
  updateRobotConfiguration: action((state, payload) => {
    state.robotsConfiguration = payload;
  }),
  setRobotActivity: thunk((actions, payload, { getState }) => {
    const storeState = getState();
    const currentRobotConf = storeState.robotsConfiguration.map((robot) => {
      const currentRobot = robot;
      if (robot.id === payload.id) {
        currentRobot.previousActivity = robot.currentActivity;
        currentRobot.currentActivity = payload.value;
      }
      return currentRobot;
    });
    actions.updateRobotConfiguration(currentRobotConf);
  }),
  incrementFooCounter: action((state, payload) => {
    state.countFoo = payload + 1;
  }),
  incrementBarCounter: action((state, payload) => {
    state.countBar = payload + 1;
  }),
  incrementFooBarCounter: action((state, payload) => {
    state.countFooBar = payload + 1;
  }),
  decrementFooCounter: action((state, payload) => {
    state.countFoo = payload - 1;
  }),
  decrementBarCounter: action((state, payload) => {
    state.countBar = payload - 1;
  }),
  decrementBuyRobot: action((state, payload) => {
    state.countFoo = payload.foo - 6;
    state.countFooBar = payload.foobar - 3;
  }),
  addRobot: thunk((actions, payload, { getState }) => {
    const storeState = getState();
    const currentRobotConfLength = storeState.robotsConfiguration.length;
    const newConf = storeState.robotsConfiguration;
    newConf.push({ id: currentRobotConfLength + 1, currentActivity: '', isRobotAvailable: true, previousActivity: '' });

    actions.updateRobotConfiguration(newConf);
  }),
};


export const store = createStore<IStoreModel>(model);
