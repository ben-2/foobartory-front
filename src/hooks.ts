import { createTypedHooks } from 'easy-peasy';
import { IStoreModel } from './store';

const { useStoreActions, useStoreState, useStoreDispatch, useStore } = createTypedHooks<IStoreModel>();

export {
  useStoreActions,
  useStoreState,
  useStoreDispatch,
  useStore
};