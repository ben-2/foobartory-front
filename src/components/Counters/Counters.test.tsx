import { createStore, StoreProvider } from 'easy-peasy';
import { Counters } from './Counters';
import { store } from '../../store';
import { mount } from 'enzyme';
import { useStoreState } from '../../hooks';

describe('<Counters />', () => {
  it('match snapshot', () => {
    const initialStateForTest = {
      countFoo: 0
    };
    const storeTest = createStore(store, { initialState: initialStateForTest });
    const wrapper = mount(
      <StoreProvider store={storeTest}>
        <Counters />
      </StoreProvider>)
    expect(wrapper).toMatchSnapshot();
  });
});
