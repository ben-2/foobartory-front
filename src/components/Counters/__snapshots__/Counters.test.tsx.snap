// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Counters /> match snapshot 1`] = `ReactWrapper {}`;
